### PR TITLE
Migrate UserGuide.md from markdown to asciidoc

### PR DESCRIPTION
**Pull request by limmlingg**
Part of #271

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@limmlingg submitted v1 for review.

* [v1 1/1] [Migrate UserGuide\.md from markdown to asciidoc ](https://github.com/se-edu/addressbook-level4/pull/435/commits/617897860510a197563b2a54dd7f4ef9b73d57c8)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/cc22ed642b34f15798facec1bddd597aa8c0c395...617897860510a197563b2a54dd7f4ef9b73d57c8))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/435/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+* Items with `...` after them can have multiple instances.
+* Parameters can be in any order.
+====
+
+=== Viewing help : `help`
+
+Format: `help`
+
+[TIP]
+Help is also shown if you enter an incorrect command e.g. `abcd`
```
Comment (Zhiyuan-Amos) [inline docs/UserGuide.adoc]: nooooo

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+* *Add* `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` +
+e.g. `add James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+* *Clear* : `clear`
+* *Delete* : `delete INDEX` +
+e.g. `delete 3`
+* *Find* : `find KEYWORD [MORE_KEYWORDS]` +
+e.g. `find James Jake`
+* *List* : `list` +
+e.g.
```
Comment (Zhiyuan-Amos) [inline docs/UserGuide.adoc]: nooooooo

**Comment by yamgent** [comment]
Could you post another CanIHasReview for archive & interdiff purpose?

**Comment by CanIHasReview** [comment]
# v2

@limmlingg submitted v2 for review.

* [v2 1/1] [Migrate UserGuide\.md from markdown to asciidoc ](https://github.com/se-edu/addressbook-level4/pull/435/commits/a348cea75911ec48771000c7e51e5e1e02951819)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/cc22ed642b34f15798facec1bddd597aa8c0c395...a348cea75911ec48771000c7e51e5e1e02951819)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/83e545441f66e10861cf1d880d0f776b641f6b62~1...83e545441f66e10861cf1d880d0f776b641f6b62))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/435/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

File: README.md [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 
 #### Site Map
-* [User Guide](docs/UserGuide.md)
+* [User Guide](docs/UserGuide.adoc)
 * [Developer Guide](docs/DeveloperGuide.md)
```
Comment (eugenepeh) [inline README.md]: Seems like the DeveloperGuide link is not updated. Opened issue #442.

Comment (eugenepeh) [inline README.md]: Oops didn't realized it has already been resolved in the main branch. I apologize for that.

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+== Quick Start
+
+.  Ensure you have Java version `1.8.0_60` or later installed in your Computer.
+
+[NOTE]
+Having any Java 8 version is not enough. +
+This app will not work with earlier versions of Java 8.
+
+[start=2]
```
Comment (pyokagan) [inline docs/UserGuide.adoc]: The generated HTML looks something like this (simplified):
```html
<ol>
<li> Ensure you have Java version... </li>
</ol>
<note>Having any Java 8 version...</note>
<ol>
<li>Download the latest...
```
Note that the `[NOTE]` is not part of the list. Besides, it not being imo semantically correct, it is cutting off the list, which is why we end up having workarounds like `[start=2]`.

Maybe use [list continuations](http://asciidoc.org/chunked/ch17.html#X15) instead? e.g.
```diff
diff --git a/docs/UserGuide.adoc b/docs/UserGuide.adoc
index 0d6d31b..10d64ac 100644
--- a/docs/UserGuide.adoc
+++ b/docs/UserGuide.adoc
@@ -15,22 +15,20 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
 == Quick Start
 
 .  Ensure you have Java version `1.8.0_60` or later installed in your Computer.
-
++
 [NOTE]
 Having any Java 8 version is not enough. +
 This app will not work with earlier versions of Java 8.
 
-[start=2]
 .  Download the latest `addressbook.jar` from the link:../../../releases[releases] tab.
 .  Copy the file to the folder you want to use as the home folder for your Address Book.
-.  Double-click the file to start the app. The GUI should appear in a few seconds.
-
+.  Double-click the file to start the app. The GUI should appear in a few seconds.+
 image::Ui.png[width="790"]
-[start=5]
+
 .  Type the command in the command box and press kbd:[Enter] to execute it. +
 e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 .  Some example commands you can try:
-
++
 * *`list`* : lists all contacts
 * **`add`**`John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01` : adds a contact named `John Doe` to the Address Book.
 * **`delete`**`3` : deletes the 3rd contact shown in the current list
```

**Comment by CanIHasReview** [comment]
# v3

@limmlingg submitted v3 for review.

* [v3 1/1] [Migrate UserGuide\.md from markdown to asciidoc ](https://github.com/se-edu/addressbook-level4/pull/435/commits/f80feac7c4595429a65aa480a1e8ef9901905fa5)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/6af393367c998910784881146ffa049725bc31bd...f80feac7c4595429a65aa480a1e8ef9901905fa5)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level4/compare/ed584fe9080e60b0ff6d3af352cd97d000557a10~1...ed584fe9080e60b0ff6d3af352cd97d000557a10))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/435/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Pull request merged**